src/modules/fancyzones/editor/FancyZonesEditor/App.xaml.cs

```diff
@@ -27,46 +36,52 @@ public App()
             }
         }
 
-        private void OnStartup(object sender, StartupEventArgs e)
+        public void OnStartup(object sender, StartupEventArgs e)
         {
-            LayoutModel[] foundModel = new LayoutModel[NumMonitors];
+            FoundModel = new LayoutModel[NumMonitors];
 
             for (int setting = 0; setting < NumMonitors; setting++)
             {
                 foreach (LayoutModel model in ZoneSettings[setting].DefaultModels)
                 {
-                    if (model.Type == Settings.ActiveZoneSetLayoutType)
+                    if (model.Type == ZoneSettings[setting].ActiveZoneSetLayoutType)
                     {
                         // found match
-                        foundModel[setting] = model;
+                        FoundModel[setting] = model;
                         break;
                     }
                 }
 
-                if (foundModel == null)
+                if (FoundModel == null)
                 {
                     foreach (LayoutModel model in Settings.CustomModels)
                     {
                         if ("{" + model.Guid.ToString().ToUpper() + "}" == Settings.ActiveZoneSetUUid.ToUpper())
                         {
                             // found match
-                            foundModel[setting] = model;
+                            FoundModel[setting] = model;
                             break;
                         }
                     }
                 }
 
-                if (foundModel == null)
+                if (FoundModel == null)
                 {
-                    foundModel[setting] = ZoneSettings[setting].DefaultModels[0];
+                    FoundModel[setting] = ZoneSettings[setting].DefaultModels[0];
                 }
 
-                foundModel[setting].IsSelected = true;
+                FoundModel[setting].IsSelected = true;
             }
-
-            EditorOverlay overlay = new EditorOverlay();
-            overlay.Show();
-            overlay.DataContext = foundModel[0];
+            Overlay = new EditorOverlay[NumMonitors];
+            ActiveMonitors = new bool[NumMonitors];
+            LoadSetup();
+        }
+        public static void LoadSetup()

```

Comment:
nit: Rename to "LoadCurrentMonitorSetup" or something along these lines - the current name is too generic.